devops/docker/Dockerfile.typeannotation

```diff
@@ -0,0 +1,12 @@
+FROM fedora:27
+
+# To install mypy from pip
+RUN dnf install -y python3-devel python3-pip gcc
+
+RUN  pip3 install mypy
+
+WORKDIR /src
+VOLUME /src
+
+
+CMD ["mypy", "./securedrop", "./admin"]
```

Comment:
It would be better to have that in `devops/docker/Dockerfile.linting` so all linting operations are in one place.

Reply:
This requires a latest Python3 system. That is why I made a new container out of Fedora. Also, this mounts the directory as a VOLUME to the container, which is better as we don't have to build the actual container every time.

Reply:
I don't understand what you mean by " build the actual container every time." ?

Reply:
The linting container copies the source into the container, and then runs within that. Means, if I have made some changes, and then did not rebuild the container, it will not get those changes into the linting container.

Where as here we are just mounting the source code directory as a `VOLUME`.  But, your next comment makes it invalid. I will update the PR now.

Reply:
yeah @dachary - @kushaldas 's motivation makes sense here. Ideally we really want to be doing volume mounting.. I think honestly, I just made a change request about this. but it might make more sense to utilize circleci `machine` mode and use volume mounting everywhere instead of constantly having to rebuild the containers for linting